widgets/centreon-widget-host-monitoring/host-monitoring/src/export.php

```diff
@@ -249,84 +248,151 @@
+            } elseif ($key == 'last_state_change' || $key == 'last_hard_state_change') {
+                $value = time() - $value;
+                $value = CentreonDuration::toString($value);
+            } elseif ($key == 'check_attempt') {
+                $value = $value . '/' . $row['max_check_attempts'] . ' (' . $aStateType[$row['state_type']] . ')';
+            } elseif ($key == 'state') {
+                $value = $stateLabels[$value];
+            } elseif ($key == 'output') {
+                $value = substr($value, 0, $outputLength);
+            } elseif (($key == 'action_url' || $key == 'notes_url') && $value) {
+                if (!preg_match("/(^http[s]?)|(^\/\/)/", $value)) {
+                    $value = '//' . $value;
+                }
+
+                $value = CentreonUtils::escapeSecure($hostObj->replaceMacroInString($row['name'], $value));
+            } elseif ($key == 'criticality' && $value != '') {
+                $critData = $criticality->getData($row['criticality_id']);
+                $value = $critData['hc_name'];
             }
-
-            $value = CentreonUtils::escapeSecure($hostObj->replaceMacroInString($row['name'], $value));
-        } elseif ($key == 'criticality' && $value != '') {
-            $critData = $criticality->getData($row['criticality_id']);
-            $value = $critData['hc_name'];
+            $data[$row['host_id']][$key] = $value;
         }
-        $data[$row['host_id']][$key] = $value;
-    }
 
-    if (isset($preferences['display_last_comment']) && $preferences['display_last_comment']) {
-        $res2 = $dbb->prepare(<<<'SQL'
-            SELECT 
-                1 AS REALTIME,
-                data
-            FROM comments
-            WHERE host_id = :hostId
-                AND service_id IS NULL
-            ORDER BY entry_time DESC
-            LIMIT 1
-            SQL
-        );
-        $res2->bindValue(':hostId', $row['host_id'], \PDO::PARAM_INT);
-        $res2->execute();
-        if ($row2 = $res2->fetch()) {
-            $data[$row['host_id']]['comment'] = substr($row2['data'], 0, $commentLength);
-        } else {
-            $data[$row['host_id']]['comment'] = '-';
+        if (isset($preferences['display_last_comment']) && $preferences['display_last_comment']) {
+            $res2 = $dbb->prepare(<<<'SQL'
+                SELECT 
+                    1 AS REALTIME,
+                    data
+                FROM comments
+                WHERE host_id = :hostId
+                    AND service_id IS NULL
+                ORDER BY entry_time DESC
+                LIMIT 1
+                SQL
+            );
+            $res2->bindValue(':hostId', $row['host_id'], \PDO::PARAM_INT);
+            $res2->execute();
+            $data[$row['host_id']]['comment'] = ($row2 = $res2->fetch()) ? substr($row2['data'], 0, $commentLength) : '-';
         }
     }
+} catch (CentreonDbException $e) {
+    CentreonLog::create()->error(
+        CentreonLog::TYPE_SQL,
```

Comment:
A log is already made from CentreonDB in the sql_error.log file, you can log into centreon-web.log for the business logic :) Also your message could be more precise, indicating that this is the context of host monitoring exports.

```suggestion
        CentreonLog::TYPE_BUSINESS_LOG,
```